ZIR example (#5)
Veridise/zk-language-comparison: zir/README.md

Reviewer: For the README, can we add:
- Instructions on how to download/setup any required dependencies/links to such things
- Instructions on how to compile the circuits and run the tests

Author: I'm planning to write those down as soon as I have the steps clear

Reviewer: In line with some below nits, I'd rename this directory `zirgen` and not `zir`.

> # ZIR

Reviewer: ```suggestion
# Zirgen
```

It's a little pedantic, but according to https://github.com/risc0/zirgen, Zirgen is the compiler name and the name of the DSL.

Author: Ah, I see. I always thought the name of language was zir

> ZIR is a DSL created by the Risc0 team to support their zkvm. 

Reviewer: ```suggestion
Zirgen is a DSL created by the Risc0 team to support their zkvm. 
```

## Building zirgen 

> To build with risc0's ZIR frontend (`zirgen`) first clone their repository and then build the tool. To build the tool you will need to install [Bazel](https://bazel.build/) version 6.0.0, which is the required version at the time of writing. If this version number is no longer correct the version declared in the `.bazelversion` file inside the repo prevails. An easy way to manage different bazel versions is with [Bazelisk](https://github.com/bazelbuild/bazelisk). 

Reviewer: ```suggestion
To build with risc0's zirgen frontend (also called `zirgen`), first clone their repository and then build the tool. 
To build the tool you will need to install [Bazel](https://bazel.build/) version 6.0.0, which is the required version at the time of writing. 
If this version number is no longer correct the version declared in the `.bazelversion` file inside the repo prevails. 
An easy way to manage different bazel versions is with [Bazelisk](https://github.com/bazelbuild/bazelisk). 
```
